presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -168,6 +168,9 @@ public ConnectorSplitSource getSplits(
 
         // get table metadata
         HiveMetadata metadata = (HiveMetadata) hiveTransactionManager.get(transaction);
+        if (metadata == null) {
+            throw new PrestoException(HIVE_TRANSACTION_NOT_FOUND, format("Transaction not found: %s", transaction));
```

Comment:
Is this expected to happen after your fix (i.e. the error code `HIVE_TRANSACTION_NOT_FOUND` is never expected to be presented to user? ) ?  If it's not expected to happen shouldn't this be a `checkState` ?

Reply:
It still can happen. It is just instead of throwing a NPE it is better to throw something more meaningful.